include/olg_scheduler/scheduler.hpp

```diff
@@ -109,17 +98,35 @@ class Event : private cavl::Node<Event<TimePoint>>
     [[nodiscard]] std::optional<TimePoint> getDeadline() const noexcept { return deadline_; }
 
 protected:
+    using Tree = cavl::Tree<Event>;
+    using cavl::Node<Event>::remove;
+
     /// The event is not automatically scheduled, it must be explicitly scheduled by the user.
-    explicit Event(cavl::Tree<Event>& tree) : tree_(tree) {}
+    Event() = default;
+
+    /// The event can be safely destroyed at any time.
+    /// It will be automatically canceled and removed from the event loop.
+    virtual ~Event()
```

Comment:
GCC for some reason is not happy if I remove `virtual` from `~Event` - looks like false positive.